python/hail/plot/plots.py

```diff
@@ -133,3 +231,102 @@ def qq(pvals):
     bound = max(max(exp), max(obs)) * 1.1
     p.line([0, bound], [0, bound], color='red')
     return p
+
+
+@typecheck(pvals=expr_float64, locus=nullable(expr_locus()), title=nullable(str),
+           size=int, hover_fields=nullable(dictof(str, expr_any)))
+def manhattan(pvals, locus=None, title=None, size=4, hover_fields=None):
+    """Create a Manhattan plot. (https://en.wikipedia.org/wiki/Manhattan_plot)
+
+    Parameters
+    ----------
+    pvals : :class:`.Float64Expression`
+        P-values to be plotted.
+    locus : :class:`.LocusExpression`
+        Locus values to be plotted.
+    title : str
+        Title of the plot.
+    size : int
+        Size of markers in screen space units.
+    hover_fields: Dict[str, :class:`.Expression`]
+        Dictionary of field names and values to be shown in the HoverTool of the plot.
+
+    Returns
+    -------
+    :class:`bokeh.plotting.figure.Figure`
+    """
+    def get_contig_index(x, starts):
+        left = 0
+        right = len(starts) - 1
+        while left <= right:
+            mid = (left + right) // 2
+            if x < starts[mid]:
+                if x >= starts[mid - 1]:
+                    return mid - 1
+                right = mid
+            elif x >= starts[mid+1]:
+                left = mid + 1
+            else:
+                return mid
+
+    pvals = -hail.log10(pvals)
+
+    if locus is None:
+        locus = pvals._indices.source.locus
+
+    if hover_fields is None:
+        hover_fields = {}
+
+    hover_fields['locus'] = hail.str(locus)
+    res = hail.tuple([locus.global_position(), pvals, locus.contig, hail.struct(**hover_fields)]).collect()
+    hf_struct = [point[3] for point in res]
+
+    for key in hover_fields:
+        hover_fields[key] = [item[key] for item in hf_struct]
+
+    x = [point[0] for point in res]
+    y = [point[1] for point in res]
+    label = [point[2] for point in res]
```

Comment:
Something is wrong here - [2] is the contig

Reply:
Where is contig used? I'm on mobile and it's not easy to see

Reply:
@tpoterba yeah the contig is the label, and points[3] is the hf_struct. The label is then passed in to scatter() for coloring.

Reply:
We can figure out the contig from the binary search though, without needing to collect more data. Hmm

Reply:
@tpoterba I'll move the call to scatter() to after the binary search